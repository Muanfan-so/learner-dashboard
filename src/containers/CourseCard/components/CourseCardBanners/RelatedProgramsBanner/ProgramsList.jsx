/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

export const iconMap = {
  XSeries: 'xseries-icon',
  MicroMasters: 'micromasters-icon',
  MicroBachelors: 'microbachelors-icon',
};

export const ProgramsList = ({ programs, isCollapse }) => programs.map((program) => (
  <span key={program.programUrl} className={isCollapse ? 'd-inline' : 'row'}>
    <span className="col text-ellipsis w-0 px-1">
      <span
        className={`pgn__icon pgn__icon__inline d-inline-block mr-2 ${
          iconMap[program.programType] || ''
        }`}
      />
      <a href={program.programUrl}>{program.title}</a>
      {isCollapse && <span className="ml-2">â€¢</span>}
    </span>
  </span>
));

ProgramsList.propTypes = {
  programs: PropTypes.arrayOf(
    PropTypes.shape({
      // bannerImgSrc: PropTypes.string,
      // logoImgSrc: PropTypes.string,
      // numberOfCourses: PropTypes.number,
      programType: PropTypes.string,
      programUrl: PropTypes.string,
      // provider: PropTypes.string,
      title: PropTypes.string,
    }).isRequired,
  ),
  isCollapse: PropTypes.bool,
};

export default ProgramsList;
