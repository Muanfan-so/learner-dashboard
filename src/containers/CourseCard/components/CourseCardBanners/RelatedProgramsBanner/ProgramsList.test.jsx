import { shallow } from 'enzyme';

import { ProgramsList, iconMap } from './ProgramsList';

describe('ProgramsList', () => {
  const programs = [
    ...Object.keys(iconMap).map((key) => ({
      title: `program title${key}`,
      programUrl: `http://example.com/${key}`,
      programType: key,
    })),
    // undefined programType
    {
      title: 'undefined programType',
      programUrl: 'http://example.com/undefined-programType',
      programType: undefined,
    },
  ];

  it('renders correctly', () => {
    const wrapper = shallow(<ProgramsList programs={programs} />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.length).toEqual(programs.length);
    wrapper.forEach((el) => expect(el.hasClass('row')).toEqual(true));
  });

  it('renders correctly with collapse', () => {
    const wrapper = shallow(<ProgramsList programs={programs} isCollapse />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.length).toEqual(programs.length);
    wrapper.forEach((el) => expect(el.hasClass('d-inline')).toEqual(true));
  });
});
