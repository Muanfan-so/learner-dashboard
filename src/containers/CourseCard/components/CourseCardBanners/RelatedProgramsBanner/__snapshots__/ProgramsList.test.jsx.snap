// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramsList renders correctly 1`] = `
Array [
  <span
    className="row"
    key="http://example.com/XSeries"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 xseries-icon"
      />
      <a
        href="http://example.com/XSeries"
      >
        program titleXSeries
      </a>
    </span>
  </span>,
  <span
    className="row"
    key="http://example.com/MicroMasters"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 micromasters-icon"
      />
      <a
        href="http://example.com/MicroMasters"
      >
        program titleMicroMasters
      </a>
    </span>
  </span>,
  <span
    className="row"
    key="http://example.com/MicroBachelors"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 microbachelors-icon"
      />
      <a
        href="http://example.com/MicroBachelors"
      >
        program titleMicroBachelors
      </a>
    </span>
  </span>,
  <span
    className="row"
    key="http://example.com/undefined-programType"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 "
      />
      <a
        href="http://example.com/undefined-programType"
      >
        undefined programType
      </a>
    </span>
  </span>,
]
`;

exports[`ProgramsList renders correctly with collapse 1`] = `
Array [
  <span
    className="d-inline"
    key="http://example.com/XSeries"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 xseries-icon"
      />
      <a
        href="http://example.com/XSeries"
      >
        program titleXSeries
      </a>
      <span
        className="ml-2"
      >
        •
      </span>
    </span>
  </span>,
  <span
    className="d-inline"
    key="http://example.com/MicroMasters"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 micromasters-icon"
      />
      <a
        href="http://example.com/MicroMasters"
      >
        program titleMicroMasters
      </a>
      <span
        className="ml-2"
      >
        •
      </span>
    </span>
  </span>,
  <span
    className="d-inline"
    key="http://example.com/MicroBachelors"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 microbachelors-icon"
      />
      <a
        href="http://example.com/MicroBachelors"
      >
        program titleMicroBachelors
      </a>
      <span
        className="ml-2"
      >
        •
      </span>
    </span>
  </span>,
  <span
    className="d-inline"
    key="http://example.com/undefined-programType"
  >
    <span
      className="col text-ellipsis w-0 px-1"
    >
      <span
        className="pgn__icon pgn__icon__inline d-inline-block mr-2 "
      />
      <a
        href="http://example.com/undefined-programType"
      >
        undefined programType
      </a>
      <span
        className="ml-2"
      >
        •
      </span>
    </span>
  </span>,
]
`;
